### ISSUE TITLE
Exception handling

### ISSUE THREAD

**Issue by teucer** [omnilib/aiosqlite#55]
### Description

Database is locked when there is an exception. 

PS: I am using  [databases](https://github.com/encode/databases) where I don't get access to underlying connection and cursor. 

### Details
Minimal reproducible example:

```python

import aiosqlite

async def test():
    db = await aiosqlite.connect("data.db")

    create_sql = """
    CREATE TABLE edits (
    id INTEGER NOT NULL,
    category TEXT NOT NULL,
    question TEXT NOT NULL,
    answer TEXT NOT NULL,
    action VARCHAR(6) NOT NULL,
    CONSTRAINT pk_edits PRIMARY KEY (id),
    CONSTRAINT ck_edits_action CHECK (action  in ('INSERT', 'UPDATE', 'DELETE')))
    """

    data = {
        "category": "cat1",
        "question": "hello",
        "answer": "world",
        "action": "USDATE",
    }

    columns = ", ".join(data.keys())
    sql = "INSERT INTO edits ({}) VALUES {}".format(columns, tuple(data.values()))
    cursor = await db.execute(create_sql)
    cursor = await db.execute(sql)
    await cursor.close()
    await db.commit()
    await db.close()
```
start ipython and execute `await test()`.

I am correctly getting the `IntegrityError: CHECK constraint failed: ck_edits_action`. But then I cannot exit ipython and accessing from another process raises  db locked error. 

* OS: Windows 7
* Python version: 3.6.8
* aiosqlite version: 0.11.0
* Can you repro on master? NA
* Can you repro in a clean virtualenv? Yes

**Comment by rdn4github** [comment]
An even more concise example:

- OS: Windows 10 Enterprise
- Python Version: 3.7.4
- aiosqlite version: 0.12.0
- Can you repro on master? NA
- Can you repro in a clean virtualenv? Yes

```
import asyncio
import aiosqlite


async def test():
    db = await aiosqlite.connect("/cannot/open/this/database.db")
    await db.close()

asyncio.run(test())
```

Results:

```
returning exception unable to open database file   
Traceback (most recent call last):
File "C:\Users\<redacted>\AppData\Local\Programs\Python\Python37\lib\site-packages\aiosqlite\core.py", line 153, in run
result = function()
File "C:\Users\<redacted>\AppData\Local\Programs\Python\Python37\lib\site-packages\aiosqlite\core.py", line 314, in connector
return sqlite3.connect(loc, **kwargs)
sqlite3.OperationalError: unable to open database file
```
And then the program hangs.

**Comment by jreese** [comment]
Connection failures resulting in hung threads (@rdn4github's issue) should be resolved in master branch, and I added test cases to catch any regressions. 

As for the original issue, I believe this is mostly working "as designed", and the hanging thread is because you are not using the context manager, and also not catching the exception and closing the connection in your exception handler. An exception due to bad inserts should not close the connection/thread, and due to Python reference counting, there's no good way to detect that the reference has gone out of scope to close it automatically (except by using it as a context manager).

Recommended usage would be something like this to correctly close the db when exceptions occur, without needing to explicitly catch those exceptions:

```python
async def test():
    async with aiosqlite.connect("data.db") as db:

        create_sql = """
        CREATE TABLE edits (
        id INTEGER NOT NULL,
        category TEXT NOT NULL,
        question TEXT NOT NULL,
        answer TEXT NOT NULL,
        action VARCHAR(6) NOT NULL,
        CONSTRAINT pk_edits PRIMARY KEY (id),
        CONSTRAINT ck_edits_action CHECK (action  in ('INSERT', 'UPDATE', 'DELETE')))
        """

        data = {
            "category": "cat1",
            "question": "hello",
            "answer": "world",
            "action": "USDATE",
        }

        columns = ", ".join(data.keys())
        sql = "INSERT INTO edits ({}) VALUES {}".format(columns, tuple(data.values()))
        cursor = await db.execute(create_sql)
        cursor = await db.execute(sql)
        await cursor.close()
        await db.commit()
```

**Comment by rdn4github** [comment]
This also fixed #35
